docs/src/custom-transactions/transaction-builders.md

```diff
@@ -0,0 +1,85 @@
+
+## Creating a custom transaction
+
+Here is an example outlining some of the features of the transaction builders.
+
+In this scenario, we have a predicate that holds some bridged asset with ID **bridged_asset_id**. It releases it's locked assets if the transaction sends **ask_amount** of the base asset to the **receiver** address:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx_receiver}}
+```
+
+Our goal is to create a transaction that will use our hot wallet to transfer the **ask_amount** to the **receiver** and then send the unlocked predicate assets to a second wallet that acts as our cold storage.
+
+Let's start by instantiating a builder. Since we don't plan to deploy a contract, the `ScriptTransactionBuilder` is the appropriate choice:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx}}
+```
+
+Next, we need to define transaction inputs of the base asset that sum up to **ask_amount**. We also need transaction outputs that will assing those assets to the predicate address and thereby unlock it. The methods `get_asset_inputs_for_amount` and `get_asset_outputs_for_amount` can help with that. We need to specify the asset ID, the target amount and the target address:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx_io_base}}
+```
+
+Let's repeat the same process but this time for transferring the assets held by the predicate to our cold storage:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx_io_other}}
+```
+
+We combine all of the inputs and outputs and set them on the builder:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx_io}}
+```
+
+As we have used coins that require a signature, we sign the transaction builder with:
+
+```rust,ignore
+{{#include ../../../examples/cookbook/src/lib.rs:custom_tx_sign}}
+```
+
+> **Note** The signature is not created until the transaction is finalized with `build(&provider)`
+
+We need to do one more thing before we stop thinking about transaction inputs. Executing the transaction also incurs a fee that is paid with the base asset. Our base asset inputs need to be large enough so that the total amount covers the transaction fee and any other operations we are doing. The Account trait let's us use `adjust_for_fee()` for adjusting the transactions inputs if needed to cover the fee. The second argument to `adjust_for_fee()` is the total amount of the base asset that we expect our tx to spend regardless of fees. In our case, this is the **ask_amount** we are transferring to the predicate.
```

Comment:
```suggestion
We need to do one more thing before we stop thinking about transaction inputs. Executing the transaction also incurs a fee that is paid with the base asset. Our base asset inputs need to be large enough so that the total amount covers the transaction fee and any other operations we are doing. The Account trait lets us use `adjust_for_fee()` for adjusting the transaction inputs if needed to cover the fee. The second argument to `adjust_for_fee()` is the total amount of the base asset that we expect our tx to spend regardless of fees. In our case, this is the **ask_amount** we are transferring to the predicate.
```